packages/core/src/common/message-service.ts

```diff
@@ -72,15 +73,35 @@ export class MessageService {
         return this.client.showMessage({ type, text });
     }
 
-    newProgress(message: ProgressMessageArguments): Promise<ProgressToken | undefined> {
-        return this.client.newProgress(message);
-    }
-
-    stopProgress(progress: ProgressToken): Promise<void> {
-        return this.client.stopProgress(progress);
+    async showProgress(message: ProgressMessage, onDidCancel?: () => void): Promise<Progress> {
+        const id = this.newProgressId();
+        const cancellationSource = new CancellationTokenSource();
+        const report = (update: ProgressUpdate) => {
+            this.client.reportProgress(id, update);
+        };
+        let clientMessage = message;
+        if (message.options && message.options.cancelable) {
+            if (!(message.actions || []).some(a => a === 'Cancel')) {
+                clientMessage = { ...message, actions: [...(message.actions || []), 'Cancel'] };
+            }
+        }
+        const result = this.client.showProgress(id, clientMessage, cancellationSource.token);
+        result.then(value => {
+            if (message.options && message.options.cancelable && typeof onDidCancel === 'function' && value === 'Cancel') {
+                onDidCancel();
+            }
+        });
+        return {
+            id,
+            cancel: () => cancellationSource.cancel(),
```

Comment:
This didn't close the notification for me

Reply:
@vinokurig, I've cleaned up too much in the end. the following changes add closing of messages. 

BTW: this is how I test it on the BE
```ts
    async test() {
        const didCancelCallback = () => this.messageService.warn('did cancel!', { timeout: 0 });
        const progress = await this.messageService.showProgress({ text: 'clock', options: { cancelable: true } }, didCancelCallback);
        const tick = () => progress.report({ increment: 1, value: String(new Date().toISOString()) });
        const intervall = setInterval(tick, 1000);
        const stop = () => { clearInterval(intervall); progress.cancel(); };
        setTimeout(stop, 10000);
    }
```

![2018-11-20 17 50 01](https://user-images.githubusercontent.com/914497/48789519-8d7d3f00-eced-11e8-9921-20caf1dc2cc9.gif)